src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -411,41 +358,45 @@ private SafeProcessHandle GetProcessHandle()
         /// <summary>Get the minimum and maximum working set limits.</summary>
         private void GetWorkingSetLimits(out IntPtr minWorkingSet, out IntPtr maxWorkingSet)
         {
-            SafeProcessHandle handle = null;
-            try
+            using (SafeProcessHandle handle = GetProcessHandle(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION))
             {
-                handle = GetProcessHandle(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION);
-
-                // GetProcessWorkingSetSize is not exposed in coresys.  We use 
-                // GetProcessWorkingSetSizeEx instead which also returns FLAGS which give some flexibility
-                // in terms of Min and Max values which we neglect.
                 int ignoredFlags;
                 if (!Interop.Kernel32.GetProcessWorkingSetSizeEx(handle, out minWorkingSet, out maxWorkingSet, out ignoredFlags))
-                {
                     throw new Win32Exception();
-                }
-            }
-            finally
-            {
-                ReleaseProcessHandle(handle);
             }
         }
 
         private string GetMainWindowTitle()
         {
             IntPtr handle = MainWindowHandle;
-            if (handle == (IntPtr)0)
-            {
+            if (handle == IntPtr.Zero)
                 return string.Empty;
-            }
 
-            int length = Interop.User32.GetWindowTextLength(handle) * 2;
+            int length = Interop.User32.GetWindowTextLengthW(handle);
 
             if (length == 0)
+            {
+#if DEBUG
+                // We never used to throw here, want to surface possible mistakes on our part
+                int error = Marshal.GetLastWin32Error();
+                Debug.Assert(error == 0, $"Failed GetWindowTextLengthW(): { new Win32Exception(error).Message }");
+#endif
                 return string.Empty;
+            }
 
             StringBuilder builder = new StringBuilder(length);
-            Interop.User32.GetWindowText(handle, builder, builder.Capacity);
+            length = Interop.User32.GetWindowTextW(handle, builder, builder.Capacity + 1);
+
+            if (length == 0)
+            {
+#if DEBUG
+                // We never used to throw here, want to surface possible mistakes on our part
+                int error = Marshal.GetLastWin32Error();
+                Debug.Assert(error == 0, $"Failed GetWindowTextW(): { new Win32Exception(error).Message }");
+#endif
+            }
+
+            builder.Length = length;
```

Comment:
Because StringBuilder doesn't include the "hidden" null in it's capacity- it is a weird detail that ends up in almost everyone being off by one when P/Invoking. GetWindowTextLength returns the length in characters sans null (which may or may not be more than needed). GetWindowText needs space for the null, which StringBuilder has.

I'm passing Capacity + 1 because that is the right value- it is coming straight from the "buffer". (They are both ok, but that is right.)

I want to pull all of our interop StringBuilders in the future- they over allocate and are tricky to understand.